Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs

```diff
@@ -112,8 +114,10 @@ public void ExitDisposals(EntityUid uid, DisposalHolderComponent? holder = null,
                 if (duc != null)
                     duc.Container.Insert(entity, EntityManager, xform, meta: meta);
                 else
+                {
                     _xformSystem.AttachToGridOrMap(entity, xform);
-
+                    _throwing.TryThrow(entity, _random.NextVector2(0.1f), 0.1f);
+                }
```

Comment:
Shouldn't this throw away from point of exit

Reply:
Most disposal chutes point straight up, don't they?

Reply:
Is that not what this is doing? It's trying to slightly throw the entity in a random direction.

Reply:
I think I see what you're saying now, (if the pipe is pointing to the left, the object should continue traveling to the left upon exit) but the object will be in the air for such a short amount of time, you wont even see the direction it's traveling.

Reply:
> I think I see what you're saying now, (if the pipe is pointing to the left, the object should continue traveling to the left upon exit) but the object will be in the air for such a short amount of time, you wont even see the direction it's traveling.

I think they're actually referring to the direction of the disposal unit. The normal ones don't have a direction, but in SS13 there are a variant which do. You can probably just tack on a float of how much you want them to be flung in the direction the unit is facing and just add the random vector to that for variance.

Reply:
this happens both on trunks and exiting without a trunk
also i think this needs `, user: entity` at the end to make the egg actually take damage for whatever reason

even in space the random direction is negligible so its fine
using pipe direction also has the caveat of having to add the grid rotation since throw uses map direction

Reply:
Okay so I can use `holder.PreviousDirection.ToVec()` to get the direction things were traveling. If its going to the right this value is X=1, Y=0. I don't want to pass that directly in, because it throws the entity towards 1,0. I want to throw the entity its current location.X + 1.

My question now is how do I get the location of an entity? 🙂

Reply:
previous is wrong for bends use current i believe

also thats the direction not a position
direction needs to be made relative to the map not grid

Reply:
I tried using `holder.CurrentDireciton.ToVec()` but its throwing this:
```
[FATL] unhandled: System.IndexOutOfRangeException: Index was outside the bounds of the array.
```

Not sure what's up with that... maybe because its out of the tube the current direction doesn't exist. Either way... piggy backing off of this

> this happens both on trunks and exiting without a trunk
> 
> even in space the random direction is negligible so its fine using pipe direction also has the caveat of having to add the grid rotation since throw uses map direction

Let's just go with a small random direction. The entity being flushed is in the air for a fraction of a second, so you cant even see the direction. the point is to splat things, not send them flying.